src/hooks/use-check-light-curate.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react'
+import { useWeb3Context } from 'web3-react'
+import { ethers } from 'ethers'
+import { useParams } from 'react-router'
+import _gtcr from '../assets/abis/GeneralizedTCR.json'
+
+const useCheckLightCurate = () => {
+  const { library, active } = useWeb3Context()
+  const { tcrAddress } = useParams<{ tcrAddress: string }>()
+
+  const [isLightCurate, setLightCurate] = useState<boolean>(false)
+
+  useEffect(() => {
+    const checkLightCurate = async () => {
+      try {
+        if (!active) return
+        const tcr = new ethers.Contract(tcrAddress, _gtcr, library)
```

Comment:
You should change this hook completely to make a query with Apollo instead. You can figure out if a contract is Classic or Light by querying if a `lregistry` exists. Doing this will resolve the query faster + it will be cached. Otherwise, you need to do this query every time + web3 queries are slow.